### ISSUE TITLE
RB 360 guitar model XBGTS2-A not working in fortnite but pc and app detects it

### ISSUE THREAD

**Issue by Believeinsteve92** [InvoxiPlayGames/FestivalInstrumentMapper#43]
**Fixed, see below.**

I do have ignore controller input off, if I don't turn hidhide on I can control in game with it such as moving my character oddly enough but it doesn't detect it as a guitar. 

Its possible its something related to rb4.app as I used the guitar with this adapter from roll limitless developer on my xbox series x, but the only thing I can think of is I'm manually selecting the xbox 360 wireless receiver adapter. Driver version 10.0.22621.3374, from driver date 3/21/2024.

EDIT: Doesn't seem to work with wireless model 822153 as well. This is the wireless receiver I use if it makes a difference; https://www.amazon.com/gp/product/B076GZFLR3/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

Fixed: Update. I was able to get my wireless RB2 guitar working with a third party receiver. Its important to know the change I think that made the difference for me was the driver I used for wireless receiver. I don't know how others got their drivers installed but I'll explain how I had to do mine.

Open device manager, go to Xbox 360 Peripherals, and then Xbox 360 wireless receiver for windows (might say something else for you). Right click, properties, click driver tab, then update driver. Click browse my computer for drivers, then let me pick from a list of available drivers on my computer. Here is where I went wrong. Make sure "Show compatible hardware" checkbox is checked. Mine then only has one option, "Xbox 360 wireless receiver for windows. If you have anything else, it didn't seem to work for me. For context, I was using rb4.app to use this same guitar through an adapter on roll limitless website to have it talk to my xbox through my pc. It will say "This driver is not digitally signed!" but thats okay. 

Once you reboot your pc, turn on remapper and good to go. Another thing to note is I was using fortnite on "D" drive but moved it to C drive after hearing about this. I also mapped it in HidHide manually myself just in case. Hopefully this helps some.

**Comment by Believeinsteve92** [comment]
It seems to be related to the driver, I installed the default one and now it detects my regular 360 wireless controller which it wasn't before changing the driver. But with Hidhide off, the camera spins as its tied to the whammy bar axis. When I turn hidhide on for instrument mapper, its no longer detected. When I look in hidhide configuration client it shows "Controller (XBOX 360 For Windows)" as the connected controller. Is it supposed to say guitar?

**Comment by InvoxiPlayGames** [comment]
I'm using a third-party wireless 360 dongle and not having this issue - when you say "it's no longer detected", do you mean the controller isn't detected by Fortnite, or isn't detected by FestivalInstrumentMapper?

**Comment by Believeinsteve92** [comment]
When I map instrument, the games controls no longer respond on my guitar. The exception being the xbox 360 guide button turns my camera left for some reason. 

What third party wireless 360 dongle do you have if its different than the one I ordered?

**Comment by TheNathannator** [comment]
> The exception being the xbox 360 guide button turns my camera left for some reason.

This is most likely Steam's default desktop configuration for gamepads, make sure Steam isn't running or that Steam Input is disabled. By default, holding the guide button will make the right stick control the mouse, and Xbox 360 whammy/tilt is mapped to the right stick.

You should also try restarting your PC after configuring the hiding, to make sure that anything else which already sees the controller will no longer see it after the reboot.

**Comment by Believeinsteve92** [comment]
I did ensure that. I've restarted my pc multiple times changing the driver after it boots back up I force close steam.

**Comment by TheNathannator** [comment]
Not sure then, need more information about your setup. I'm not seeing any issues when I run things under a test rig of mine, although I don't play Fortnite to be able to test there. Just not sure where to start with this issue.

**Comment by Believeinsteve92** [comment]
Wait, I'm confused. How do you know it's working for you if you aren't able to test on fortnite? What do you consider "not seeing any issues?" The point of this app is to remap it so fortnite recognizes the guitar controller as a rb4/riffmaster guitar? Does it have functions outside of fortnite?

**Comment by Believeinsteve92** [comment]
Update. I was able to get my wireless RB2 guitar working with a third party receiver. Its important to know the change I think that made the difference for me was the driver I used for wireless receiver. I don't know how others got their drivers installed but I'll explain how I had to do mine.

Open device manager, go to Xbox 360 Peripherals, and then Xbox 360 wireless receiver for windows (might say something else for you). Right click, properties, click driver tab, then update driver. Click browse my computer for drivers, then let me pick from a list of available drivers on my computer. Here is where I went wrong. Make sure "Show compatible hardware" checkbox is checked. Mine then only has one option, "Xbox 360 wireless receiver for windows. If you have anything else, it didn't seem to work for me. For context, I was using rb4.app to use this same guitar through an adapter on roll limitless website to have it talk to my xbox through my pc. It will say "This driver is not digitally signed!" but thats okay. 

Once you reboot your pc, turn on remapper and good to go. Another thing to note is I was using fortnite on "D" drive but moved it to C drive after hearing about this. I also mapped it in HidHide manually myself just in case. Hopefully this helps some.

**Comment by Believeinsteve92** [comment]
Update. I was able to get my wireless RB2 guitar working with a third party receiver. Its important to know the change I think that made the difference for me was the driver I used for wireless receiver. I don't know how others got their drivers installed but I'll explain how I had to do mine.

Open device manager, go to Xbox 360 Peripherals, and then Xbox 360 wireless receiver for windows (might say something else for you). Right click, properties, click driver tab, then update driver. Click browse my computer for drivers, then let me pick from a list of available drivers on my computer. Here is where I went wrong. Make sure "Show compatible hardware" checkbox is checked. Mine then only has one option, "Xbox 360 wireless receiver for windows. If you have anything else, it didn't seem to work for me. For context, I was using rb4.app to use this same guitar through an adapter on roll limitless website to have it talk to my xbox through my pc. It will say "This driver is not digitally signed!" but thats okay. 

Once you reboot your pc, turn on remapper and good to go. Another thing to note is I was using fortnite on "D" drive but moved it to C drive after hearing about this. I also mapped it in HidHide manually myself just in case. Hopefully this helps some.
